packs/tfe_fdo_nomad/templates/tfe.nomad.tpl

```diff
@@ -0,0 +1,220 @@
+        timeout  = [[ .tfe_fdo_nomad.health_check_timeout | quote ]]
+      }
+    }
+
+    task "tfe-task" {
+      driver = "docker"
+
+      identity {
+         env = true
+      }
+
+       template {
+        destination = "/secrets/key.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .key.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/cert.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .cert.Value }}
+{{- end -}}
+EOF
+      }
+      template {
+        destination = "/secrets/bundle.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .bundle.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/nomad_ca_cert.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .nomad_ca.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/nomad_cert.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .nomad_cert.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/nomad_cert_key.pem"
+        change_mode = "restart"
+        splay       = "60s"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+  {{ base64Decode .nomad_cert_key.Value }}
+{{- end -}}
+EOF
+      }
+
+      template {
+        destination = "/secrets/secrets.env"
+        env         = true
+        change_mode = "restart"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+TFE_LICENSE                             = {{ .tfe_license }}
+TFE_DATABASE_PASSWORD                   = {{ .db_password }}
+TFE_OBJECT_STORAGE_S3_SECRET_ACCESS_KEY = {{ .s3_secret_key }}
+TFE_REDIS_PASSWORD                      = {{ .redis_password }}
+TFE_ENCRYPTION_PASSWORD                 = {{ .tfe_encryption_password }}
+TFE_IMAGE_REGISTRY_PASSWORD             = {{ .tfe_image_registry_password }}
+{{- end -}}
+EOF
+      }
+
+      config {
+         image = [[ .tfe_fdo_nomad.tfe_image | quote ]]
+         auth {
+            # User Input is mandatory.
+            username = [[ .tfe_fdo_nomad.tfe_image_registry_username | quote ]]
+            # User Input is mandatory.
+            password       = "${TFE_IMAGE_REGISTRY_PASSWORD}"
+            server_address = [[ .tfe_fdo_nomad.tfe_image_server_address | quote ]]
+         }
```

Comment:
Not blocking for this PR: we should look at whether we can require a cred helper here so that the registry password isn't hard-coded in the jobspec. That's sort of hard to workaround in a generic way though.

Reply:
Thanks @tgross, env variables are go to way to setup registry passwords in TFE. We have tried to incorporate the same here as well.